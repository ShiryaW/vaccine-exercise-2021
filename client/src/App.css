.app {
  text-align: left;
  margin: 0 auto;
}

.ag-theme-alpine {
  margin: 0 auto;
  width: 100%;
  height: 96vh;
  overflow-x: hidden;
}

.dropdown {
  display: flex;
  height: 4vh;
}

.dropdown p {
  padding: 0 25px;
}

select {
  display: flex;
  padding: 12px;
  font-size: 16px;
  width: 250px;
}