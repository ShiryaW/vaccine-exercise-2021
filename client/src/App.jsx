import "./App.css";
import React, { Component } from "react";
import { AgGridReact } from "ag-grid-react";

import "ag-grid-community/dist/styles/ag-grid.css";
import "ag-grid-community/dist/styles/ag-theme-alpine.css";

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      rows: [
        {
          id: "6da3a8cf-c923-4c77-8f80-c69c935fe1df",
          orderNumber: 1,
          responsiblePerson: "Joonatan Siloma",
          healthCareDistrict: "KYS",
          vaccine: "Antiqua",
          injections: 4,
          arrived: "2021-01-11T08:59:28.642790Z",
        },
        {
          id: "1251aa6c-ebaf-4e33-89d3-d6f210497b94",
          orderNumber: 2,
          responsiblePerson: "Tarvo Puro",
          healthCareDistrict: "TAYS",
          vaccine: "Antiqua",
          injections: 4,
          arrived: "2021-01-10T01:29:26.642846Z",
        },
        {
          id: "c00e2610-5bd9-4f84-9597-1e7febfae62c",
          orderNumber: 4,
          responsiblePerson: "Linda Väisälä",
          healthCareDistrict: "HYKS",
          vaccine: "Antiqua",
          injections: 4,
          arrived: "2021-01-08T05:33:37.642901Z",
        },
      ],
    };

    this.columnDefs = [
      {
        headerName: "ID",
        field: "id",
      },
      {
        headerName: "Order number",
        field: "orderNumber",
      },
      {
        headerName: "Responsible person",
        field: "responsiblePerson",
      },
      {
        headerName: "Healthcare district",
        field: "healthCareDistrict",
      },
      {
        headerName: "Vaccine",
        field: "vaccine",
      },
      {
        headerName: "Injections",
        field: "injections",
      },
      {
        headerName: "Arrived on",
        field: "arrived",
      },
    ];
  }

  componentDidMount() {
    window.addEventListener("resize", this.handleResize);
  }

  handleResize(e) {
    //TODO: get this to work
    console.log(e);
    this.gridApi && this.gridApi.sizeColumnsToFit();
  }

  handleRowsFetch = async () => {
    await fetch(`/orders?brand=antiqua`, {
      method: "GET",
      headers: {
        Accept: "application/json",
      },
    })
      .then((res) => {
        return res.json();
      })
      .then((data) => {
        console.log(data);
      });
  };

  onGridReady({ api }) {
    this.gridApi = api;
    this.gridApi.sizeColumnsToFit();
  }

  render() {
    const gridOptions = {
      rowData: this.state.rows,
      columnDefs: this.columnDefs,
      onGridReady: this.handleRowsFetch,
    };

    return (
      <div className="Vaccines">
        <div className="ag-theme-alpine">
          <AgGridReact gridOptions={gridOptions} />
        </div>
      </div>
    );
  }
}

export default App;
