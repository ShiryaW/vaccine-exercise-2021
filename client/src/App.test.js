import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import App from "./App";
import React from "react";
import { shallow } from "enzyme";

describe("App", () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it("renders the UI", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
