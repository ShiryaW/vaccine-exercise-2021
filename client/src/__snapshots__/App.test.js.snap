// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the UI 1`] = `
<div
  className="app"
>
  <div
    className="grid"
  >
    <Dropdown
      buttonChecked={false}
      groupBy="healthcare district"
      onChange={[Function]}
      onToggle={[Function]}
    />
    <div
      className="ag-theme-alpine"
    >
      <AgGridReact
        disableStaticMarkup={false}
        gridOptions={
          Object {
            "animateRows": true,
            "columnDefs": Array [
              Object {
                "field": "id",
                "headerName": "ID",
              },
              Object {
                "field": "orderNumber",
                "headerName": "Order number",
                "sortable": true,
                "unSortIcon": true,
              },
              Object {
                "field": "responsiblePerson",
                "headerName": "Responsible person",
                "sortable": true,
                "unSortIcon": true,
              },
              Object {
                "field": "healthCareDistrict",
                "headerName": "Healthcare district",
                "rowGroup": false,
              },
              Object {
                "field": "injections",
                "headerName": "Injections",
              },
              Object {
                "field": "arrived",
                "headerName": "Arrived on",
                "sortable": true,
                "unSortIcon": true,
              },
            ],
            "defaultColDef": Object {
              "resizable": true,
            },
            "onGridReady": [Function],
          }
        }
        legacyComponentRendering={false}
        maxComponentCreationTimeMs={1000}
        rowData={Array []}
      />
    </div>
    <Footer
      totalItems={0}
    />
  </div>
  <div
    className="sidebar"
  >
    <Calendar
      maxDate={+275760-09-13T00:00:00.000Z}
      maxDetail="month"
      minDate={0000-12-31T22:20:11.000Z}
      minDetail="century"
      onChange={[Function]}
      returnValue="start"
      showNavigation={true}
      showNeighboringMonth={true}
      view="month"
    />
  </div>
</div>
`;
